### ISSUE TITLE
Bays-by-back/ loss function

### ISSUE THREAD

**Issue by bashhwu** [pawni/BayesByHypernet#2]
Hi @pawni,

Do not you think that the factor (1/#of batches) should be added to  KL_loss, not to error loss?
please refer to Alex Graves, 2011: Practical Variational Inference for Neural Networks

Why are you averaging the error loss (cross_entropy)? 
According to the original paper, error loss should be summed (not averaged) over the batch.

**Comment by pawni** [comment]
Hi,

The scaling can be added on either side (as long as it then is inverted).
It is possible to take the mean but then you only scale with the number of data samples - I have a full refactor I am working on and will fix those issues in the code then.

The mean principle still holds though.
